src/thunderbots/software/ai/hl/stp/evaluation/robot.cpp

```diff
@@ -14,3 +16,18 @@ bool Evaluation::robotOrientationWithinAngleThresholdOfTarget(const Point positi
     Angle diff_orientation   = orientation.minDiff(target_orientation);
     return diff_orientation < threshold;
 }
+
+bool Evaluation::robotHasPossession(const Ball ball, const Robot robot)
+{
+    // The actual vector to the ball from the center of the robot
+    Vector robot_center_to_ball = ball.position() - robot.position();
+
+    // Calculate the ideal vector from the robot to the ball for the robot to have
+    // possession.
+    Angle orientation          = robot.orientation();
+    double expected_position_x = orientation.cos() * DIST_TO_FRONT_OF_ROBOT_METERS;
```

Comment:
Just a tip, you can create `Vectors` directly from angles to avoid having to do the trig yourself. You can then set/normalize the length of the vector with the `norm()` function.

Reply:
Oh that's easier. Thanks